extension/src/background/messageListener/popupMessageListener.ts

```diff
@@ -92,49 +93,65 @@ export const popupMessageListener = (request: Request) => {
     return nonHwKeyIds[0] || "";
   };
 
+  // in lieu of using KeyManager, let's store hW data in local storage
+  // using schema:
+  // "hw:<G account>": {
+  //   publicKey: "",
+  //   bipPath: "",
+  // }
   const _storeHardwareWalletAccount = ({
     publicKey,
     hardwareWalletType,
+    bipPath,
   }: {
     publicKey: string;
     hardwareWalletType: WalletType;
+    bipPath: string;
   }) => {
     const mnemonicPhrase = mnemonicPhraseSelector(store.getState());
-    const allAccounts = allAccountsSelector(store.getState());
+    let allAccounts = allAccountsSelector(store.getState());
 
     const keyId = `${HW_PREFIX}${publicKey}`;
     const keyIdListArr = getKeyIdList();
     const accountName = `${hardwareWalletType} ${
       keyIdListArr.filter((k: string) => k.indexOf(HW_PREFIX) !== -1).length + 1
     }`;
-    keyIdListArr.push(keyId);
 
-    localStorage.setItem(KEY_ID_LIST, JSON.stringify(keyIdListArr));
+    if (keyIdListArr.indexOf(keyId) === -1) {
```

Comment:
also making sure the same ledger account cant be imported twice. I noticed we can still for regular accounts though, I'll create a ticket for that

Reply:
https://github.com/stellar/freighter/issues/556

Reply:
Ah yeah, that's def an issue. Thanks for creating the ticket!